x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/pages/coverage_overview/filters_panel.tsx

```diff
@@ -5,42 +5,188 @@
+import { DashboardFilterButtonComponent } from './shared_components/dashboard_filter_button';
 import { CoverageOverviewLegend } from './shared_components/dashboard_legend';
+import {
+  formatRuleFilterOptions,
+  getInitialRuleStatusFilterOptions,
+  getInitialRuleTypeFilterOptions,
+} from './helpers';
 import * as i18n from './translations';
+import { ruleStatusFilterDefaultOptions, ruleTypeFilterDefaultOptions } from './constants';
 
 export interface CoverageOverviewFiltersPanelProps {
-  setShowExpandedCells: (arg: boolean) => void;
-  showExpandedCells: boolean;
+  isLoading: boolean;
 }
 
 const CoverageOverviewFiltersPanelComponent = ({
-  setShowExpandedCells,
-  showExpandedCells,
+  isLoading,
 }: CoverageOverviewFiltersPanelProps) => {
+  const {
+    dispatch,
+    state: { showExpandedCells, filter },
+  } = useCoverageOverviewDashboardContext();
+
+  const setShowExpandedCells = useCallback(
+    (value: boolean): void => {
+      dispatch({
+        type: 'setShowExpandedCells',
+        value,
+      });
+    },
+    [dispatch]
+  );
+
+  const setRuleStatusFilter = useCallback(
+    (value: CoverageOverviewRuleActivity[]): void => {
+      dispatch({
+        type: 'setRuleStatusFilter',
+        value,
+      });
+    },
+    [dispatch]
+  );
+
+  const setRuleTypeFilter = useCallback(
+    (value: CoverageOverviewRuleSource[]): void => {
+      dispatch({
+        type: 'setRuleTypeFilter',
+        value,
+      });
+    },
+    [dispatch]
+  );
+
+  const setRuleSearchFilter = useCallback(
+    (value: string): void => {
+      dispatch({
+        type: 'setRuleSearchFilter',
+        value,
+      });
+    },
+    [dispatch]
+  );
+
   const handleExpandCellsFilterClick = () => setShowExpandedCells(true);
   const handleCollapseCellsFilterClick = () => setShowExpandedCells(false);
 
+  const ruleStatusFilterInitalValue = useMemo(
+    () => getInitialRuleStatusFilterOptions(filter),
+    [filter]
+  );
+  const ruleTypeFilterInitalValue = useMemo(
+    () => getInitialRuleTypeFilterOptions(filter),
+    [filter]
+  );
+
+  const [ruleStatusFilterOptions, setRuleStatusFilterOptions] = useState(
+    ruleStatusFilterInitalValue
+  );
+
+  const [ruleTypeFilterOptions, setRuleTypeFilterOptions] = useState(ruleTypeFilterInitalValue);
+
+  const handleRuleStatusFilterOnChange = useCallback(
+    (options) => {
+      setRuleStatusFilterOptions(options);
+      const formattedOptions = formatRuleFilterOptions<CoverageOverviewRuleActivity>(options);
+      setRuleStatusFilter(formattedOptions);
+    },
+    [setRuleStatusFilter]
+  );
+
+  const handleRuleTypeFilterOnChange = useCallback(
+    (options) => {
+      setRuleTypeFilterOptions(options);
+      const formattedOptions = formatRuleFilterOptions<CoverageOverviewRuleSource>(options);
+      setRuleTypeFilter(formattedOptions);
+    },
+    [setRuleTypeFilter]
+  );
+
+  const handleRuleStatusFilterOnClear = useCallback(() => {
```

Comment:
Basically handling of filters could be done a dedicated components, potentially reused in the future. Instead of converting back and forth via `formatRuleFilterOptions` and `getInitialRuleStatusFilterOptions` a component can accept props with all available values (or hardcode it under the hood) and selected values and does the necessary tricks under the hood to display the selector. So outside it may look like

```
<RuleActivity selected={['enabled', 'disabled']} onChange={handleRuleActivityChange} />
```

`CoverageOverviewDashboardContext` should have the state containing filters set in default values initially.